src/components/WorkflowEngine/engine.ts

```diff
@@ -0,0 +1,169 @@
+import { performHttpReq, HttpRequestParams } from '~utils/http'
+import { formatError } from '~utils/onfidoApi'
+import type { StepConfig } from '~types/steps'
+import type {
+  WorkflowResponse,
+  OutcomeStepKeys,
+  GetWorkflowFunc,
+  CompleteWorkflowFunc,
+  GetFlowStepFunc,
+  WorkflowStepConfig,
+} from './utils/WorkflowTypes'
+
+export interface EngineInterface {
+  getOutcomeStep(workflow: WorkflowResponse | undefined): OutcomeStepKeys
+  getWorkflow: GetWorkflowFunc
+  completeWorkflow: CompleteWorkflowFunc
+  getWorkFlowStep: GetFlowStepFunc
+}
+
+export type EngineProps = {
+  token: string | undefined
+  workflowRunId: string | undefined
+  workflowServiceUrl: string
+}
+
+export type EngineConfiguration = {
+  delay: number
+}
+
+export class Engine implements EngineInterface {
+  public engineProps: EngineProps
+
+  constructor(engineProps: EngineProps) {
+    this.engineProps = engineProps
+  }
+
+  getOutcomeStep = (workflow: WorkflowResponse): OutcomeStepKeys => {
+    return !workflow.has_remaining_interactive_tasks
+      ? 'complete'
+      : workflow.outcome
+      ? 'pass'
+      : 'reject'
+  }
+
+  getWorkflow: GetWorkflowFunc = async (): Promise<WorkflowResponse> => {
+    const { token, workflowRunId, workflowServiceUrl } = this.engineProps
+    if (!token) {
+      throw new Error('token not provided')
+    }
+
+    return new Promise((resolve, reject) => {
+      try {
+        const requestParams: HttpRequestParams = {
+          method: 'GET',
+          contentType: 'application/json',
+          endpoint: `${workflowServiceUrl}/workflow_runs/${workflowRunId}`,
+        }
+
+        if (token) {
```

Comment:
It cannot run without a token anyway. Therefore the if can be removed and the token header placed into line 55.